containers/popv/context/main.py

```diff
@@ -49,10 +63,19 @@ def prepare_query(
         )
         model_path = self.find_model_dir(options["models_dir"], organ)
         reference_data = scanpy.read_h5ad(reference_data_path)
+        data, var_names = self.normalize_var_names(data, options)
+        # data.var_names = t.cast(t.Any, var_names)
+        updated_data = self.add_model_genes(
+            data,
+            model_path,
+            options["data_genes_column"],
+        )
+        updated_data.var_names_make_unique()  # for resolving duplicate index bug
+        updated_data.X = numpy.rint(updated_data.X)  # for resolving non integer bug
+        updated_data.write_h5ad("lookat_2.h5ad")
```

Comment:
That looks weird

Reply:
Yes! should remove the write h5ad line. Thanks for pointing out